### DISCUSSION TITLE
pyscript equivalent to HA yaml "mode: queued"

### DISCUSSION THREAD

**Discussion by dy-nz** [custom-components/pyscript#478] [General] [answered]
I'm new to HA, and even newer to pyscript. 
After fighting with automations, bash scripts and nc/netcat so that HA can talk to my devices (two of which are multizone amplifiers, one a receiver), I've just discovered pyscript and import telnet. Relief :-)

I have one significant thing I can't find a pyscript solution for. I use the HA yaml "mode: queued" to ensure that only one HA action is  telneting with one of my devices at any one time. It is simple to use. Is there a pyscript equivalent please ?

Thanks.
David

**Comment by craigbarratt** [comment, marked as answer]
First, you should be using [`asynctelnet`](https://pypi.org/project/asynctelnet/) instead of the vanilla `telnet` module to avoid blocking.

To ensure your telnet I/O is serialized among multiple callers, you should use [`asyncio.Lock()`](https://docs.python.org/3/library/asyncio-sync.html#lock), or one of the other async synchronization primitives.  See #340 for an example.
